src/gtc/cuir/cuir_codegen.py

```diff
@@ -507,6 +506,9 @@ def ctype(symbol: str) -> str:
         """
     )
 
+    def __init__(self):
+        super().__init__(traits.SymbolTableTrait.add_symtable)
+
```

Comment:
If the previsitors are defined per class instead of per instance, I think it would be enough to write something like :
```suggestion
previsitors: ClassVar[Optional[...]] = (traits.SymbolTableTrait.add_symtable,)
```